File: fixtures/scale/typescript/large/src/checks/CheckFnvhash.ts
import { Fnvhash } from '../ciphers/Fnvhash';
import { FnvhashSpec } from '../specs/FnvhashSpec';
import { CorpusHash } from '../corpus/CorpusHash';

export class CheckFnvhash {
  static run(): boolean {
    const spec = FnvhashSpec.get();
    for (const text of CorpusHash.texts()) {
      const first = Fnvhash.digest(text);
      const second = Fnvhash.digest(text);
      if (first !== second || first.length !== 8) return false;
    }
    return spec.category === 'hash';
  }
}

File: fixtures/scale/lua/large/checks/check_fnvhash.lua
local fnvhash = require("ciphers.fnvhash")
local fnvhash_spec = require("specs.fnvhash_spec")
local corpus_hash = require("corpus.corpus_hash")

local function check_fnvhash()
  local spec = fnvhash_spec.get()
  for _, text in ipairs(corpus_hash.corpus_hash()) do
    local first = fnvhash.fnvhash_digest(text)
    local second = fnvhash.fnvhash_digest(text)
    if first ~= second or #first ~= 8 then return false end
  end
  return spec.category == "hash"
end

return { check_fnvhash = check_fnvhash }
